### PR TITLE
Basecamp: fix issues with basecamp recipes

### LINKED ISSUE #32643: Basecamp skill requirement for blacksmith forge is inconsistent
# Describe the bug

`Building a forge and kiln is our first task.` says that `Skills used: none`, but when i pick the NPC for the job, it tells me it needs `fabrication 5`

0.D-5847-ga4acee28f4-dirty

### LINKED ISSUE #34776: 0,0,0 map tile full of items: soil,tools,steel barrels
# Describe the bug
Yesterday experimental:
2019-10-15-171446_1920x1080_scrot

# Steps To Reproduce

1. Teleport to 0,0,0 in any of your latest experimental saves.

# Expected behavior
Usual random tile.

# Versions and configuration
- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 18.04.3 LTS; Release: 18.04; Codename: bionic; 
- Game Version: 0.D-8520-gcb7938e [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    More Survival Tools [More_Survival_Tools],
    Disable Religious Texts [no_religious_Texts],
    No Antique Firearms [no_olg_guns]
]

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Bugfixes "Basecamp: fix issues with basecamp recipes"```

#### Purpose of change
Fixes #34776
Fixes #32643

The basecamp construction requirement autocalc code is not supposed to save the tinymap at 0,0,0 used to determine construction requirements, but does.  Work around the issue by creating the tinymap at 0,0,-9 where most players are unlikely to find it.

Basecamp recipes that use "skill_used" and "difficulty" did not correctly report the skill requirements during mission selection, even though the requirements showed up during companion selection.  Fix that issue and generally clean up handling of skill requirements during companion selection.

#### Describe the solution
Create the fake_map used to set up construction requirements at 0,0,-9 instead of 0,0,0.

Print recipe requirements that use "skill_used" and "difficulty" instead of the the skills_required map.  For recipes that use skills_required, print each skill and whether the NPC meets the requirements in the companion selection menu.  Confirm that NPCs meet the requirements and don't let them be selected if they do not.

#### Describe alternatives you've considered
skill_used and difficulty should be just rolled into skills_required, possibly, but I'm not sure about the additional effects in other part of the code.

#### Testing
Started a new game and teleported to 0,0,0 and  confirmed that it was a normal field like the fields all around it.  Then teleported to 0,0,-9 and confirmed there was the remnants of the fake map.

Created a faction camp, and verified that complex buildings like wattle-and-daub displayed both skills (Fabrication 4, Survival 3) in both the mission description and companion selection menu.  Confirmed that single skill recipes like the Garage showed correctly in both menus.  Confirmed that I couldn't select an NPC that didn't have the required skills and that I could select an NPC that could.

### CODE REVIEW

**Comment by esotericist** [comment]
I'm not in a position to look closely at the actual c++ code right now, but as a stop-gap measure to fix the user-facing bug on the way to 0.E, this makes sense (given that it'd basically require a max depth lab to happen to spawn at `0,0`).

Hopefully the actual problem can be tracked down at some point, after we're past the release.

File: src/mission_companion.cpp [hunk 1]
```diff
@@ -1949,51 +1958,64 @@ npc_ptr talk_function::companion_choose( const std::string &skill_tested, int sk
...
 
-    skill_id skill_tested_id( skill_tested );
     int x = 0;
-    for( auto &e : available ) {
-        std::string npc_entry;
+    std::string menu_header = left_justify( _( "Who do you want to send?" ), 51 );
+    if( required_skills.empty() ) {
+        menu_header += _( "[ COMBAT : SURVIVAL : INDUSTRY ]" );
+    }
+    for( npc_ptr &e : available ) {
```
Comment (davidpwbrown) [inline src/mission_companion.cpp]: Could be const?

File: src/recipe.cpp [hunk 2]
```diff
@@ -438,7 +438,17 @@ bool recipe::has_byproducts() const
 std::string recipe::required_skills_string( const Character *c, bool print_skill_level ) const
 {
     if( required_skills.empty() ) {
-        return _( "<color_cyan>none</color>" );
+        if( difficulty == 0 ) {
+            return _( "<color_cyan>none</color>" );
+        } else {
+            const auto player_skill = c ? c->get_skill_level( skill_used ) : 0;
```
Comment (davidpwbrown) [inline src/recipe.cpp]: int?

File: src/mission_companion.cpp (hunk 1 above)
Comment (davidpwbrown) [inline src/mission_companion.cpp]: could be const?

File: src/recipe.cpp (hunk 2 above)
Comment (mlangsdorf) [reply src/recipe.cpp]: Fixed, thanks.

File: src/mission_companion.cpp (hunk 1 above)
Comment (mlangsdorf) [reply src/mission_companion.cpp]: Thanks!

**Pull request merged**